### PR TITLE
FluxC: Media pagination

### LINKED ISSUE #5177: Fluxc - Media Integration - Paging is broken in media browser
Pagination is broken in media browser

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #5177. This PR implements the new paging changes introduced in the [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/322).

P.S: Please let me know before merging this in so I can revert the commit that changes `build.gradle`: b911006e516e5f4bb11390463d9ebb9516785c12

/cc @aforcier & @maxme

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaItemFragment.java [hunk 1]
```diff
@@ -416,9 +414,4 @@ private void copyUrlToClipboard() {
             ToastUtils.showToast(getActivity(), R.string.error_copy_to_clipboard);
         }
     }
-
-    @Subscribe(threadMode = ThreadMode.MAIN)
-    public void onMediaChanged(MediaStore.OnMediaChanged event) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaItemFragment.java]: This crash if we don't remove `mDispatcher.register()` / `mDispatcher.unregister()` calls, and we can also remove `mDispatcher` altogether.

Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaItemFragment.java]: Fixed in 603516462c8475c3437ff9d057e73608a0a39e1c.

I forgot about how that worked. I think it's better to remove the dispatcher from places where we don't use it.

File: WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java [hunk 2]
```diff
@@ -793,36 +779,25 @@ private void fetchAllMedia(int offset) {
...
+        if (!mIsRefreshing) {
             mIsRefreshing = true;
             updateEmptyView(EmptyViewMessageType.LOADING);
             setRefreshing(true);
             mGridAdapter.setRefreshing(true);
 
-            // TODO: figure out how to integrate `auto` to callback
-            MediaFilter fetchFilter = new MediaFilter();
-            fetchFilter.offset = mMediaFetchOffset;
-            fetchFilter.number = NUM_PER_FETCH;
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java]: We can drop the `NUM_PER_FETCH` constant now.

Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/media/MediaGridFragment.java]: Fixed in bfedc49921897536f73625fd66bdb3ce262dacae

**Comment by maxme** [comment]
Looks good to me, I tried with different sites (.com, Jetpack, self hosted) in the media browser and the editor (wp media picker).

Note: travis 🔴 - I restarted the builds.

**Comment by aforcier** [comment]
Good to go by me as well 👍

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**